staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -84,9 +88,8 @@ func parseIP(str string) (net.IP, error) {
 	if err != nil {
 		return nil, err
 	}
-	//TODO add ipv6 support
 	if len(bytes) != net.IPv4len {
-		return nil, fmt.Errorf("only IPv4 is supported")
+		return nil, fmt.Errorf("invalid IPv4 address in route")
 	}
 	bytes[0], bytes[1], bytes[2], bytes[3] = bytes[3], bytes[2], bytes[1], bytes[0]
```

Comment:
Could you elaborate on how to do that in a way that is more obvious that the above? I've been digging into this and can't seem to see an easy way to convert the hex byte slice to big endian hex byte slice:
```
	b, err := hex.DecodeString(str)
        ...
	var b2 uint32
	binary.Read(bytes.NewBuffer(b), binary.LittleEndian, &b2)
	b3 := make([]byte, 4)
	binary.BigEndian.PutUint32(b3, b2)
        return net.IP(b3), nil
```